src/app/shared/carousel/carousel.ts

```diff
@@ -10,17 +10,23 @@ import {
   ViewChild,
   ViewEncapsulation,
 } from '@angular/core';
+import {FocusableOption, FocusKeyManager} from '@angular/cdk/a11y';
 
 
 @Directive({
   selector: '[carousel-item]',
 })
-export class CarouselItem {
+export class CarouselItem implements FocusableOption {
+  @HostBinding('attr.role') readonly role = 'listitem';
   @HostBinding('style.width.px') width = this.carousel.itemWidth;
   @HostBinding('tabindex') tabindex = '-1';
 
   constructor(readonly carousel: Carousel, readonly element: ElementRef) {
   }
+
+  focus(): void {
+    this.element.nativeElement.focus({preventScroll: true});
```

Comment:
Why prevent scroll? I just tested and found that I could sometimes get the focus to be off screen

Reply:
The automatic scrolling conflicts with `_shiftItems` when you get to the end of the visible cards and the next focus is not yet visible, do you know if theres a way to just disable horizontal scroll?

Reply:
I would make the images `aria-hidden="true"` since they're purely decorative